emcc.py

```diff
@@ -1632,9 +1643,12 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
       if opt_level >= 2:
         JSOptimizer.queue += ['asmLastOpts']
 
-      if shared.Settings.FINALIZE_ASM_JS: JSOptimizer.queue += ['last']
+      if shared.Settings.EMTERPRETIFY:
```

Comment:
also here, i worry about the complexity - we should not need to check for intersections of modes like emterpretify and DFE.